app/controllers/products_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ProductsController < ApplicationController
+  before_action :set_nav
+  before_action :set_product, except: [:new, :create]
```

Comment:
That being said, it's important to consider whether this pattern might make your code a little harder to parse later on. If someone has forgotten that this happens, they might be puzzled by seemingly undeclared references to `@product`. This isn't a super big deal when your app is relatively simple, but it can cause headaches when apps get more complex, especially for other programmers.